### DISCUSSION TITLE
Right click drag and contextmenu

### DISCUSSION THREAD

**Discussion by mjanulaitis** [punker76/gong-wpf-dragdrop#381] [Q&A]
I am trying to right click and drag within a TreeView which I have done in plenty of .Net Forms apps in the past, however, the context menu keeps trying to display.  I just noticed Windows Explorer actually has the same bug.  Sometimes it works but many times Explorer fights with the right click drag and the context menus.  I have implemented my ContextMenus on each tree node so I can't, to my knowledge, simple disable the context menu on the TreeView.  Does anyone have experience trying to get this control to work when right click dragging over items that have ContextMenues?

**Comment by punker76** [comment]
@mjanulaitis Normally the ContextMenu will be opened if you release the mouse button. I added a test menu to the first TreeView sample and set the `CanDragWithMouseRightButton` to true and can drag&drop the items.

The TreeViewItem style for the test:

```xaml
<Style x:Key="BoundTreeViewItemStyle" TargetType="{x:Type TreeViewItem}">
    <Setter Property="ContextMenu">
        <Setter.Value>
            <ContextMenu>
                <MenuItem Header="Test 1" />
            </ContextMenu>
        </Setter.Value>
    </Setter>
</Style>
```

The TreeView:

```xaml
<TreeView x:Name="LeftBoundTreeView"
          dd:DragDrop.IsDragSource="True"
          dd:DragDrop.IsDropTarget="True"
          dd:DragDrop.UseDefaultDragAdorner="True"
          dd:DragDrop.UseDefaultEffectDataTemplate="True"
          dd:DragDrop.CanDragWithMouseRightButton="True"
          dd:DragDrop.SelectDroppedItems="True"
          ItemContainerStyle="{StaticResource BoundTreeViewItemStyle}"
          ItemsSource="{Binding Data.TreeCollection1}">
    <TreeView.ItemTemplate>
        <HierarchicalDataTemplate ItemsSource="{Binding Children}">
            <TextBlock Margin="2" Text="{Binding Caption}" />
        </HierarchicalDataTemplate>
    </TreeView.ItemTemplate>
</TreeView>
```

**Reply by mjanulaitis** [reply]
@punker76 Thanks, I'm not sure if I tried that option, however, I will take a look at it again when I complete my current features.
